### PR TITLE
Add test to cover spatial aggregation bug

### PR DESCRIPTION
**Pull request by cijothomas**
For Observablecounters, tests with tags were missing. This adds more coverage. Also added a test which is skipped now, until spatially aggregating the ObservableCounter is supported.

### CODE REVIEW

**Review by utpilla — APPROVED** [review]

File: test/OpenTelemetry.Tests/Metrics/MetricAPITest.cs [hunk 1]
```diff
@@ -382,6 +384,131 @@ public void ObservableCounterAggregationTest(bool exportDelta)
...
+            var metricPoint2 = metricPoints[1];
+            Assert.Equal(10, metricPoint2.GetSumLong());
+            ValidateMetricPointTags(tags2, metricPoint2.Tags);
+
+            var metricPoint3 = metricPoints[2];
+            Assert.Equal(10, metricPoint3.GetSumLong());
+            ValidateMetricPointTags(tags3, metricPoint3.Tags);
+        }
+
+        [Theory(Skip = "Known issue.")]
```
Comment (utpilla) [inline test/OpenTelemetry.Tests/Metrics/MetricAPITest.cs:451]: Do we have an issue created for this? We could add that information here.

Comment (cijothomas) [reply test/OpenTelemetry.Tests/Metrics/MetricAPITest.cs:451]: will create and link.

Comment (martincostello) [reply test/OpenTelemetry.Tests/Metrics/MetricAPITest.cs:451]: 4 years have passed so the chances might not be good, but do you remember what this issue is/was @cijothomas? I stumbled on the skip while doing some refactoring.

Comment (cijothomas) [reply test/OpenTelemetry.Tests/Metrics/MetricAPITest.cs:451]: The issue still exists, and the test added in this PR shows the issue. I think I missed creating an issue in this repo.
It originated from spec discussion https://github.com/open-telemetry/opentelemetry-specification/issues/1874

**Pull request merged**